### PR TITLE
Added systemtap testsuite_v3

### PR DESCRIPTION
**Pull request by poojabs7**
Addressed review comments.

Signed-off-by: Pooja <pooja@linux.vnet.ibm.com>

### CODE REVIEW

File: toolchain/systemtap.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                     destination_dir=self.srcdir)
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+        build.make(self.srcdir, extra_args='install')
+        self.test_dir = os.path.join(self.srcdir, "testsuite")
+        os.chdir(self.test_dir)
+        # Run a simple systemtap script to make sure systemtap and the
+        # kernel debuginfo packages are correctly installed
+        script_result = process.run("PATH=%s/bin:$PATH stap -v /bin/true "
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: can process.system use here  as it it provide exit code ?

File: toolchain/systemtap.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                      "kernel debuginfo package may be missing")
+
+    def test(self):
+        make_option = self.params.get('make_option', default='installcheck')
+        build.make(self.srcdir, extra_args=make_option)
+        # path of the log file. self.srcdir/testsuite/systemtap.sum
+        failed_tests = process.system_output("grep -w FAIL systemtap.sum",
+                                             shell=True, ignore_status=True)
+        if failed_tests:
+            self.log.info(failed_tests)
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: is showing return code is required ?

File: toolchain/systemtap.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+
+class Systemtap(Test):
+
+    def setUp(self):
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        # TODO: Add debs for Ubuntu.
+        if detected_distro.name == "Ubuntu":
+            self.skip("Skip the test")
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: can message will be proper example can be` "test is not supported in current distro so skipping test"`

File: toolchain/systemtap.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import distro
+
+
+class Systemtap(Test):
+
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: can you please write up some info about test as it help tester as what this test does ?

**Comment by poojabs7** [comment]
Opened New PR.
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/486

**Pull request closed without merging**